### PR TITLE
The next batch of symbol names

### PR DESCRIPTION
**Pull request by rocky**
These were driven by strings in the "head" parameter off of Expression running the combinatorica benchmarks. 

Alas, this is not all of them yet. 

<a href="https://gitpod.io/#https://github.com/Mathics3/mathics-core/pull/319"></a>

### CODE REVIEW

File: mathics/builtin/compilation.py [hunk 1]
```diff
@@ -166,7 +171,7 @@ def _pythonized_mathics_expr(*x):
         if cfunc is None:
             evaluation.message("Compile", "comperr", expr)
             args = Expression("List", *names)
```
Comment (TiagoCavalcante) [inline mathics/builtin/compilation.py:173]: SymbolList?

**Review by TiagoCavalcante — APPROVED** [review]
LGTM

**Review by mmatera — APPROVED** [review]
LGTM

File: mathics/builtin/compilation.py (hunk 1 above)
Comment (rocky) [reply mathics/builtin/compilation.py:173]: I will try ListExpression when I get a chance. 

In general, although the PR strays outside my immediate interest - get to a state where we are in terms of the combinatorica benchmark - I do see stuff like this a bit.  There have been specific situations, not here in particular though, where I leave something for later when it looks like things deserve more care than a simple mechanical change.

**Pull request merged**